.travis.yml

```diff
@@ -9,38 +9,45 @@ go:
   - 1.13.x
 
 os:
+  - osx
   - linux
+  - windows
 
 dist: trusty
 
 services:
   - docker
 
+before_install:
+- if [ "${TRAVIS_OS_NAME}" == "windows" ]; then choco install make; fi
+
 install:
-  - go get -u golang.org/x/lint/golint
-  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
-  - chmod +x ./cc-test-reporter
+  - ./ci/install.sh "${TRAVIS_OS_NAME}"
 
 before_script:
-  - ./cc-test-reporter before-build
+  - ./ci/before-build.sh "${TRAVIS_OS_NAME}"
 
 script:
-  - make all
+  - ./ci/build.sh "${TRAVIS_OS_NAME}"
 
 after_script:
-  - ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT -d
+  - ./ci/after-build.sh "${TRAVIS_OS_NAME}"
 
```

Comment:
I've factored these commands into scripts where it's cleaner to account for the cross-platform differences than within the CI configuration.